File: 242-valid-anagram/valid-anagram.java

class Solution {
    public boolean isAnagram(String s, String t) {
        int[] anagram = new int[26];

        if(s.length() != t.length()){
            return false;
        }

        char[] arr1 = s.toCharArray();
        char[] arr2 = t.toCharArray();

        for(char x: arr1){
            anagram[x - 'a'] += 1;
        }

        for(char y: arr2){
            anagram[y - 'a'] -=1; 
            if(anagram[y - 'a'] < 0){
                return false;
            }
        }


        return true;
    }
}

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        count = [0] * 26 

        if len(s) != len(t):
            return False

        for i in s:
            ch = ord(i) - ord('a')
            count[ch] += 1
        
        for j in t:
            ch = ord(j) - ord('a')
            count[ch] -= 1

        return all(c == 0 for c in count)
        
        
        
        
        